Makefile

```diff
@@ -12,12 +12,35 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# build options are from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/Makefile.Common
+# to make out library compatible with open telemetry
+GO_BUILD_TAGS=""
+GOTEST_OPT?= -race -timeout 300s -parallel 4 --tags=$(GO_BUILD_TAGS)
```

Comment:
So I assume this flag will help us automatically catch some go routine related race conditions, right?

Reply:
Yes. By using this flag - https://go.dev/doc/articles/race_detector - it detects race conditions. I am using it only for tests, but I would build some collectors with this flag as well to catch more of them.